### ISSUE TITLE
Computer Hardware Dataset Analysis

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/ML-Crate#503]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** : Computer Hardware Dataset Analysis
:red_circle: **Aim** : The aim of this project is to analyze the computer hardware dataset given here.
:red_circle: **Dataset** : https://www.kaggle.com/datasets/dilshaansandhu/general-computer-hardware-dataset
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Participant ID (If not, then put NA) :
- Approach for this Project :
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by zul132** [comment]
@abhisheks008 I am willing to work on this issue under JWOC. Kindly assign it to me

**Comment by abhisheks008** [comment]
Please share your approach as per the issue template.
@zul132

**Comment by zul132** [comment]
Full name : Fathima Zulaikha
GitHub Profile Link : https://github.com/zul132
Participant ID (If not, then put NA) : NA
Approach for this Project : 
For the analysis of the given Computer Hardware Dataset, I have planned to use Linear Regression, Decision Trees, K-Nearest Neighbors (KNN) and Random Forest to implement the models. After implementing the models, I will evaluate their performance using appropriate metrics and compare them to find the best-fitted algorithm for the computer hardware dataset.

What is your participant role?
I am participating under JWOC

**Comment by zul132** [comment]
> Full name : Fathima Zulaikha GitHub Profile Link : https://github.com/zul132 Participant ID (If not, then put NA) : NA Approach for this Project : For the analysis of the given Computer Hardware Dataset, I have planned to use Linear Regression, Decision Trees, K-Nearest Neighbors (KNN) and Random Forest to implement the models. After implementing the models, I will evaluate their performance using appropriate metrics and compare them to find the best-fitted algorithm for the computer hardware dataset.
> 
> What is your participant role? I am participating under JWOC

@abhisheks008

**Comment by abhisheks008** [comment]
Assigned @zul132

**Comment by zul132** [comment]
@abhisheks008 My wifi modem got damaged recently due to which I’m unable to work the ML project for this issue. Even my mobile hotspot is not getting connected to my PC. Is it possible to unassign me as the number of days allowed for Medium level (2-3 days) has already passed. My apologies for any inconveniences caused.

**Comment by abhisheks008** [comment]
Sure @zul132

**Comment by professor1412** [comment]
Full name : Kanhaiya Yadav
    GitHub Profile Link : https://github.com/professor1412
    Participant ID  : NA
    Approach for this Project :The algorithm I want to apply is Linear regression , Decision tree,KNN etc.
    What is your participant role? 
    JWoC

assingned me please.

**Comment by abhisheks008** [comment]
Assigned @professor1412

**Comment by prakharsingh-74** [comment]
As the number of days has been already passed for this issue please assign to me this if possible.

Full name : Prakhar Singh
GitHub Profile Link : https://github.com/prakharsingh-74
Participant ID  : NA
Approach for this Project :The algorithm I want to apply is Linear regression , Decision tree,KNN. I will also try to use train test spit model so see accuracy for the model.
What is your participant role? 
JWoC

**Comment by abhisheks008** [comment]
Hi @professor1412 any update on this issue, otherwise @prakharsingh-74 is ready to take up this issue.

**Comment by professor1412** [comment]
sorry for inconvenience I am unable to manage time for this . so assign it to @prakharsingh-74

**Comment by abhisheks008** [comment]
Cool. Assigned to @prakharsingh-74 under JWOC 2024.

**Comment by prakharsingh-74** [comment]
@abhisheks008 There are multiple dataset given in the link so i have to create individual model for all dataset?
![Screenshot 2024-01-30 120954](https://github.com/abhisheks008/ML-Crate/assets/148339919/29c8ce48-1375-40b7-92ed-404783f8a15a)

**Comment by abhisheks008** [comment]
You can combine the datasets altogether to make a huge dataset, you can work on that too. It's better to work with all the datasets.

**Comment by prakharsingh-74** [comment]
> You can combine the datasets altogether to make a huge dataset, you can work on that too. It's better to work with all the datasets.

I have seen recently that combining the dataaset is not possible because all these dataset has some different column and values with the reference in that particular column

**Comment by prakharsingh-74** [comment]
@abhisheks008 All the above different CSV file has different parameters for the different hardware components so it cannnot be able to merge all the file. Now, what should I do?

**Comment by abhisheks008** [comment]
Then use at least 3-4 datasets for this project which you find the most useful one.

**Comment by prakharsingh-74** [comment]
@abhisheks008 sorry to give trouble you but please unassign me this issue due to my some other works that are on my priority first. i will not be able to contribute in this issue right now for a couple of days. Thank you for giving me the oppurtunity to work on your project.

**Comment by abhisheks008** [comment]
Sure @prakharsingh-74

**Comment by shivansh-2003** [comment]
Can You Please Assign this issue under SSOC. 2024 Season 3
Shivansh Mahajan
Github:- https://github.com/shivansh-2003
Participation ID:- NA
I will do EDA of the data set by various statistical methods like IQR , Study Distribution OF Feature and Correlation Matrix.
I would train the data in Various ML model to. arrive to the better Accuracy score. I would then feed the data for Feature engineering and then train it with different machine learning models KNN , Random forest , Decision Tree , SVM and Bossting Algorithms
. I am well versed with Machine Learning you can check out my linkedin :-https://www.linkedin.com/in/shivansh-mahajan-13227824a/ and Git repository .
can u assign me with this issue @abhisheks008
Participation Role:- SSOC Season 3

**Comment by abhisheks008** [comment]
Contributions will start from June 1, 2024. Till then please have some patience.

**Comment by DarkRaiderCB** [comment]
Full name : Sanyog Mishra
GitHub Profile Link : https://github.com/DarkRaiderCB
Participant ID: NA
Approach for this Project : Will perform EDA on dataset provided using techniques like Data cleaning, categorial analysis, finding any outliers, visualisation (correlation matrix, heat maps and more), summary stats., etc. Would utilise feature engineering and use ML algorithms like Linear Regression, Decision Tree, XGBoost, KNN, etc. and will find the best performing model. Will also try to make use of Tensorflow library.
Tools to be used: Pandas, Numpy, Matplotlib, Scikit Learn, XGBoost, Tensorflow.
Resume: https://drive.google.com/file/d/1sDVtq69GJd83t4H1-EOlvHEyQc2oat1k/view?usp=drive_link
Participant Role: Contributor SSOC Season 3

**Comment by abhisheks008** [comment]
> Full name : Sanyog Mishra GitHub Profile Link : https://github.com/DarkRaiderCB Participant ID: NA Approach for this Project : Will perform EDA on dataset provided using techniques like Data cleaning, categorial analysis, finding any outliers, visualisation (correlation matrix, heat maps and more), summary stats., etc. Would utilise feature engineering and use ML algorithms like Linear Regression, Decision Tree, XGBoost, KNN, etc. and will find the best performing model. Will also try to make use of Tensorflow library. Tools to be used: Pandas, Numpy, Matplotlib, Scikit Learn, XGBoost, Tensorflow. Resume: https://drive.google.com/file/d/1sDVtq69GJd83t4H1-EOlvHEyQc2oat1k/view?usp=drive_link Participant Role: Contributor SSOC Season 3

Issue assigned to you @DarkRaiderCB

**Comment by DarkRaiderCB** [comment]
Sir pls unassign this issue as I am unable to devote time to it

**Comment by abhisheks008** [comment]
Cool @DarkRaiderCB. Thanks for the update.

**Comment by adi271001** [comment]
Full name : Aditya D
GitHub Profile Link : https://github.com/adi271001
Participant ID: NA
Approach for this Project : 
1. Data Cleaning
2. Data Preprocessing
3. EDA
4. Modelling
5. Evaluation
Participant Role: SSOC Contributor  @abhisheks008 please assign this to me if open

**Comment by adi271001** [comment]
@abhisheks008 sorry i forgot to mention i'll be using linear regression , ridge , lasso , decision tree , random forest , gradient bosst , xgboost , catboost , svm , knn , extra trees models

**Comment by abhisheks008** [comment]
Assigned @adi271001
